--- a/src/kiledocmanager.cpp
+++ b/src/kiledocmanager.cpp
@@ -1,6 +1,6 @@
 /*****************************************************************************
 *   Copyright (C) 2004 by Jeroen Wijnhout (Jeroen.Wijnhout@kdemail.net)      *
-*             (C) 2006-2010 by Michel Ludwig (michel.ludwig@kdemail.net)     *
+*             (C) 2006-2009 by Michel Ludwig (michel.ludwig@kdemail.net)     *
 *             (C) 2007 by Holger Danielsson (holger.danielsson@versanet.de)  *
 ******************************************************************************/
 
@@ -469,7 +469,7 @@ void Manager::recreateTextDocumentInfo(T
 
 	removeTextDocumentInfo(oldinfo);
 
-	emit(updateStructure(false, newinfo));
+	emit(updateStructure(true, newinfo));
 }
 
 bool Manager::removeTextDocumentInfo(TextInfo *docinfo, bool closingproject /* = false */)
@@ -605,6 +605,7 @@ KTextEditor::View* Manager::loadText(Kil
 	KTextEditor::Document *doc = createDocument(url, docinfo, encoding, mode, highlight);
 
 	m_ki->structureWidget()->clean(docinfo);
+	m_ki->structureWidget()->update(docinfo, true);
 
 	if(!text.isEmpty()) {
 		doc->setText(text);
@@ -831,6 +832,11 @@ void Manager::newDocumentStatus(KTextEdi
 	m_ki->texKonsole()->sync();
 
 	emit(documentModificationStatusChanged(doc, doc->isModified(), KTextEditor::ModificationInterface::OnDiskUnmodified));
+
+	//updatestructure if active document changed from modified to unmodified (typically after a save)
+	if (!doc->isModified()) {
+		emit(updateStructure(true, textInfoFor(doc)));
+	}
 }
 
 bool Manager::fileSaveAll(bool amAutoSaving, bool disUntitled)
@@ -925,7 +931,7 @@ bool Manager::fileSaveAll(bool amAutoSav
 	 This may look superfluos but actually it is not, in the case of multiple modified docs it ensures that the structure view keeps synchronized with the currentTextView
 	 And if we only have one masterdoc or none nothing goes wrong.
 	*/
-	emit(updateStructure(false, NULL));
+	emit(updateStructure(false,NULL));
 	m_currentlySavingAll = false;
 	return !oneSaveFailed;
 }
@@ -967,7 +973,7 @@ void Manager::fileOpen(const KUrl& url,
 		item->loadDocumentAndViewSettings();
 	}
 
-	emit(updateStructure(false, NULL));
+	emit(updateStructure(true, 0L));
 	emit(updateModeStatus());
 	// update undefined references in this file
 	emit(updateReferences(textInfoFor(realurl.toLocalFile())));
@@ -989,7 +995,6 @@ bool Manager::fileSave(KTextEditor::View
 	}
 	else {
 		bool ret = view->document()->documentSave();
-		emit(updateStructure(false, textInfoFor(view->document())));
 		return ret;
 	}
 }
@@ -1270,7 +1275,7 @@ void Manager::projectNew()
 				mapItem(docinfo, item);
 
 				//docinfo->updateStruct(m_kwStructure->level());
-				emit(updateStructure(false, docinfo));
+				emit(updateStructure(true, docinfo));
 			}
 		}
 
@@ -1502,7 +1507,7 @@ KileProject* Manager::projectOpen(const
 	kp->buildProjectTree();
 	addProject(kp);
 
-	emit(updateStructure(false, NULL));
+	emit(updateStructure(false, 0L));
 	emit(updateModeStatus());
 	// update undefined references in all project files
 	updateProjectReferences(kp);
@@ -2155,7 +2160,7 @@ void Manager::projectAddFile(QString fil
 
 	//ok, we have a project and an existing file
 	KILE_DEBUG() << "\tadd file: " << filename;
-	m_ki->viewManager()->updateStructure(false);
+	m_ki->viewManager()->updateStructure(true);
 
 	KUrl url;
 	url.setPath(filename);
--- a/src/documentinfo.cpp
+++ b/src/documentinfo.cpp
@@ -2,7 +2,7 @@
     begin                : Sun Jul 20 2003
     copyright            : (C) 2003 by Jeroen Wijnhout (Jeroen.Wijnhout@kdemail.net)
                            (C) 2005-2007 by Holger Danielsson (holger.danielsson@versanet.de)
-                           (C) 2006-2010 by Michel Ludwig (michel.ludwig@kdemail.net)
+                           (C) 2006-2009 by Michel Ludwig (michel.ludwig@kdemail.net)
  *********************************************************************************************/
 
 /***************************************************************************
@@ -169,11 +169,7 @@ KUrl Info::makeValidTeXURL(const KUrl& u
 	return newURL;
 }
 
-Info::Info() :
- m_bIsRoot(false),
- m_dirty(false),
- m_config(KGlobal::config().data()),
- documentTypePromotionAllowed(true)
+Info::Info() : m_bIsRoot(false), m_config(KGlobal::config().data()), documentTypePromotionAllowed(true)
 {
 	updateStructLevelInfo();
 }
@@ -237,17 +233,6 @@ void Info::setDocumentTypePromotionAllow
 	documentTypePromotionAllowed = b;
 }
 
-bool Info::isDirty() const
-{
-	return m_dirty;
-}
-
-void Info::setDirty(bool b)
-{
-KILE_DEBUG() << b;
-	m_dirty = b;
-}
-
 KUrl Info::url()
 {
 	return KUrl();
@@ -406,7 +391,6 @@ void Info::updateBibItems()
 
 void Info::slotCompleted()
 {
-	setDirty(true);
 	emit completed(this);
 }
 
@@ -476,7 +460,6 @@ void TextInfo::setDocument(KTextEditor::
 		connect(m_doc, SIGNAL(documentNameChanged(KTextEditor::Document*)), this, SLOT(slotFileNameChanged()));
 		connect(m_doc, SIGNAL(documentUrlChanged(KTextEditor::Document*)), this, SLOT(slotFileNameChanged()));
 		connect(m_doc, SIGNAL(completed()), this, SLOT(slotCompleted()));
-		connect(m_doc, SIGNAL(modifiedChanged(KTextEditor::Document*)), this, SLOT(makeDirtyIfModified()));
 		// this could be a KatePart bug, and as "work-around" we manually set the highlighting mode again
 		connect(m_doc, SIGNAL(completed()), this, SLOT(activateDefaultMode()));
 		setMode(m_defaultMode);
@@ -497,13 +480,6 @@ void TextInfo::detach()
 	m_doc = NULL;
 }
 
-void TextInfo::makeDirtyIfModified()
-{
-	if(m_doc && m_doc->isModified()) {
-		setDirty(true);
-	}
-}
-
 const long* TextInfo::getStatistics(KTextEditor::View *view)
 {
 	/* [0] = #c in words, [1] = #c in latex commands and environments,
@@ -1360,7 +1336,6 @@ void LaTeXInfo::updateStruct()
 	emit(doneUpdating());
 	emit(isrootChanged(isLaTeXRoot()));
 	emit(parsingCompleted());
-	setDirty(false);
 }
 
 void LaTeXInfo::checkChangedDeps()
@@ -1471,7 +1446,6 @@ void BibInfo::updateStruct()
 	}
 	emit(parsingCompleted());
 	emit(doneUpdating());
-	setDirty(false);
 }
 
 Type BibInfo::getType()
--- a/src/widgets/structurewidget.cpp
+++ b/src/widgets/structurewidget.cpp
@@ -2,7 +2,7 @@
     begin                : Sun Dec 28 2003
     copyright            : (C) 2003 by Jeroen Wijnhout (Jeroen.Wijnhout@kdemail.net
                                2005-2007 by Holger Danielsson (holger.danielsson@versanet.de)
-                               2008-2010 by Michel Ludwig (michel.ludwig@kdemail.net)
+                               2008 by Michel Ludwig (michel.ludwig@kdemail.net)
  *************************************************************************************************/
 
 /***************************************************************************
@@ -948,10 +948,10 @@ void StructureViewItem::setLabel(const Q
 
 	void StructureWidget::update(KileDocument::Info *docinfo)
 	{
-		update(docinfo, false);
+		update(docinfo, true);
 	}
 
-	void StructureWidget::update(KileDocument::Info *docinfo, bool forceParsing, bool activate /* =true */)
+	void StructureWidget::update(KileDocument::Info *docinfo, bool parse, bool activate /* =true */)
 	{
 		KILE_DEBUG() << "==KileWidget::StructureWidget::update(" << docinfo << ")=============";
 
@@ -961,24 +961,18 @@ void StructureViewItem::setLabel(const Q
 		}
 
 		m_docinfo = docinfo;
-		bool needParsing = forceParsing || m_docinfo->isDirty() || !viewExistsFor(docinfo);
+
+		bool needParsing = parse || ( ! viewExistsFor(m_docinfo) );
 
 		//find structview-item for this docinfo
 		StructureView *view = viewFor(m_docinfo);
-		if(!view) {
-			m_default->activate();
-			return;
-		}
 
 		if(needParsing) { //need to reparse the doc
 			int xtop = view->horizontalScrollBar()->value();
 			int ytop = view->verticalScrollBar()->value();
 			//KILE_DEBUG() << "\tStructure::update parsing doc";
-			// avoid flickering when parsing
-			view->setUpdatesEnabled(false);
 			view->cleanUp();
 			m_docinfo->updateStruct();
-			view->setUpdatesEnabled(true);
 			view->showReferences(m_ki);
 			view->horizontalScrollBar()->setValue(xtop);
 			view->verticalScrollBar()->setValue(ytop);
--- a/src/widgets/structurewidget.h
+++ b/src/widgets/structurewidget.h
@@ -2,7 +2,7 @@
     begin                : Sun Dec 28 2003
     copyright            : (C) 2003 by Jeroen Wijnhout (Jeroen.Wijnhout@kdemail.net)
                                2005-2007 by Holger Danielsson (holger.danielsson@versanet.de)
-                               2008-2010 by Michel Ludwig (michel.ludwig@kdemail.net)
+                               2008 by Michel Ludwig (michel.ludwig@kdemail.net)
  **************************************************************************************************/
 
 /***************************************************************************
--- a/src/documentinfo.h
+++ b/src/documentinfo.h
@@ -1,7 +1,7 @@
 /*************************************************************************************
     begin                : Sun Jul 20 2003
     copyright            : (C) 2003 by Jeroen Wijnhout (Jeroen.Wijnhout@kdemail.net)
-                           (C) 2006-2010 by Michel Ludwig (michel.ludwig@kdemail.net)
+                           (C) 2006 by Michel Ludwig (michel.ludwig@kdemail.net)
  *************************************************************************************/
 
 /***************************************************************************
@@ -159,12 +159,6 @@ public:
 	virtual bool isDocumentTypePromotionAllowed();
 	void setDocumentTypePromotionAllowed(bool b);
 
-	/**
-	 * Returns true iff new parsing is required.
-	 **/
-	bool isDirty() const;
-	void setDirty(bool b);
-
 public Q_SLOTS:
 	/**
 	 * Never call this function directly, use KileWidget::Structure::update(KileDocument::Info *, bool) instead
@@ -197,7 +191,6 @@ protected:
 	};
 
 	bool						m_bIsRoot;
-	bool						m_dirty;
 	QStringList					m_labels;
 	QStringList					m_bibItems;
 	QStringList					m_deps, m_depsPrev;
@@ -295,11 +288,8 @@ protected Q_SLOTS:
 	void slotViewDestroyed(QObject *object);
 	void activateDefaultMode();
 
-	void makeDirtyIfModified();
-
 protected:
 	KTextEditor::Document				*m_doc;
-	bool						m_dirty;
 	long						*m_arStatistics;
 	QString						m_defaultMode;
 	QHash<KTextEditor::View*, QList<QObject*> >	m_eventFilterHash;
--- a/src/kile.cpp
+++ b/src/kile.cpp
@@ -1,7 +1,7 @@
 /****************************************************************************************
     begin                : sam jui 13 09:50:06 CEST 2002
     copyright            : (C) 2003 by Jeroen Wijnhout (Jeroen.Wijnhout@kdemail.net)
-                           (C) 2007-2010 by Michel Ludwig (michel.ludwig@kdemail.net)
+                           (C) 2007-2009 by Michel Ludwig (michel.ludwig@kdemail.net)
                            (C) 2007 Holger Danielsson (holger.danielsson@versanet.de)
                            (C) 2009 Thomas Braun (thomas.braun@virtuell-zuhause.de)
  ****************************************************************************************/
