--- a/src/kile.kcfg
+++ b/src/kile.kcfg
@@ -286,7 +286,7 @@
 	<group name="Help">
 		<entry name="location" type="String">
 			<label>Location of the TeX documentation.</label>
-			<default>/usr/share/texmf/doc</default>
+                        <default>/usr/share/doc/texlive-doc</default>
 		</entry>
 		<entry name="kilerefs" type="Bool">
 			<label>Use the Kile LaTeX reference for the contextual help.</label>
