Description: fix typo and use MIME_INSTALL_DIR.
Author: Fathi Boudra <fabo@debian.org>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,7 +131,7 @@ ADD_SUBDIRECTORY( src )
 
 ########### install files ###############
 
-INSTALL( FILES x-kilepr.desktop DESTINATION share/mimelink/text )
+INSTALL( FILES x-kilepr.desktop DESTINATION ${MIME_INSTALL_DIR}/text )
 
 INSTALL( FILES kile.upd kile1.6_upd.pl kile1.7_upd.pl kile1.9_upd.pl
 kile1.9-1_upd.pl kile2.0-1_upd.pl kile2.0-2_upd.pl kile2.0-3_upd.pl
