diff -Nur kile-1.9.3/src/kile/kile.kcfg kile-1.9.3.new/src/kile/kile.kcfg
--- kile-1.9.3/src/kile/kile.kcfg	2006-11-20 10:23:18.000000000 +0100
+++ kile-1.9.3.new/src/kile/kile.kcfg	2007-05-26 13:55:46.000000000 +0200
@@ -203,7 +203,7 @@
 	<group name="Help">
 		<entry name="location" type="String">
 			<label>Location of the TeX documentation.</label>
-			<default>/usr/share/texmf/doc</default>
+			<default>/usr/share/doc/texlive-doc</default>
 		</entry>
 		<entry name="use" type="Int">
 			<label>Use the system's TeX reference for the contextual help.</label>
